### ISSUE TITLE
How can I solve the SEO problem of single page sites that are based on nuxt.js 2.15.8?

### ISSUE THREAD

**Issue by Ali-Programmer76** [nuxt/nuxt#22345]
### Environment

**In this project [https://darkob.co.ir](url) , for the SEO part of the site, we used the facilities provided by nuxt.js [that is, the property `head()`] to make the `title` and `meta description` of all the pages dynamic.**

**If we take the inspect Elements from each page of the website, we can see that it has its own `title` and `meta description`, but when we take the view page source (ctrl + U) from a page, it shows the `title` and `meta description` set in the `nuxt.config.js` file and does not change dynamically.**

**Or when we visit common SEO checker sites on the web to get information from each website page in the field of SEO; The address of each page we enter (for example, when we enter the [https://darkob.co.ir/trademark](url) address), instead of returning the` title` and `meta description` specific to this page, it returns the `title` and `meta description` that are in the `nuxt.config.js `file set.**

**I wanted to ask what is the problem? or How can I solve this problem?** 

**An example of the code that I wrote in the script section of the` trademark.vue` page:**

![Capture](https://github.com/nuxt/nuxt/assets/97738828/33ecafb1-0c93-49a5-9340-a214f0eeb9e3)

#10318

**Thank you** 

### Reproduction

[https://stackoverflow.com/questions/76770120/how-can-i-solve-the-seo-problem-of-the-following-site-which-is-based-on-nuxt-js](url)

[https://darkob.co.ir](url)

### Describe the bug

**In this project [https://darkob.co.ir](url) , for the SEO part of the site, we used the facilities provided by nuxt.js [that is, the property `head()`] to make the `title` and `meta description` of all the pages dynamic.**

**If we take the inspect Elements from each page of the website, we can see that it has its own `title` and `meta description`, but when we take the view page source (ctrl + U) from a page, it shows the `title` and `meta description` set in the `nuxt.config.js` file and does not change dynamically.**

**Or when we visit common SEO checker sites on the web to get information from each website page in the field of SEO; The address of each page we enter (for example, when we enter the [https://darkob.co.ir/trademark](url) address), instead of returning the` title` and `meta description` specific to this page, it returns the `title` and `meta description` that are in the `nuxt.config.js `file set.**

**I wanted to ask what is the problem? or How can I solve this problem?** 

**An example of the code that I wrote in the script section of the` trademark.vue` page:**

![Capture](https://github.com/nuxt/nuxt/assets/97738828/33ecafb1-0c93-49a5-9340-a214f0eeb9e3)

@danielroe 

#10318

**Thank you**

**Comment by danielroe** [comment]
Would you provide a [reproduction](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by danielroe** [comment]
If you follow the link I posted, there's some info about how to create a minimal reproduction that does not include your project code.

**Comment by Ali-Programmer76** [comment]
@danielroe  Can you give me a solution for problem #10318 ? Because my question is also related to this question.
I really don't understand your answer to issue #10318.

Thank you
